version: '3.9'

x-env-files: &default-envs
  - ./envs/docker.env

services:
  django:
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    image: mastermind_py_local_django
    depends_on:
      - postgres
    volumes:
      - .:/app
    env_file: *default-envs
    ports:
      - "8000:8000"
    command: /start
    profiles:
      - debug

  django-gunicorn:
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    image: mastermind_py_local_django
    depends_on:
      - postgres
    volumes:
      - .:/app
    env_file: *default-envs
    ports:
      - "8000:8000"
    working_dir: /app/src
    command: ["uvicorn", "main:app", "--proxy-headers", "--host", "0.0.0.0", "--port", "8000"]
    profiles:
      - app

  postgres:
    image: postgres:11.3-alpine
    init: true
    restart: on-failure
    volumes:
      - local_postgres_data:/var/lib/postgresql/data
      - local_postgres_data_backups:/backups
    ports:
      - 5432:5432
    expose:
      - "5432"
    env_file:
      - envs/docker.env
    profiles:
      - infra


volumes:
  local_postgres_data:
  local_postgres_data_backups:
